client/src/components/containers/Groups.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Group, GroupHeader, SearchButton, PreLoader } from '../presentation';
+import GroupActions from '../../actions/groupActions';
+/**
+ * Groups class which inherits the React.Component class
+ */
+class Groups extends Component {
+  /**
+   * constructor
+   * @returns {*} void
+   */
+  constructor() {
+    super();
+    this.state = {
+      users: {},
+      list: [],
+      userList: [],
+      selected: 0
+    };
+    this.groupClickHandler = this.groupClickHandler.bind(this);
+  }
+  /**
+ * componentDidMount() is called at the instance where the Groups component is being created
+ * and inserted into the DOM
+ * It fires the action that gets all the groups a user belongs to
+ * @return {*} void
+ */
+  componentDidMount() {
+    /**
+ * Fires the action for fetching all groups
+ * a user belongs to on componentDidMount
+ */
+    this.props.fetchGroups();
+    window.$('.modal-trigger').click((event) => {
+      event.preventDefault();
+      window.$('#createGroup').modal();
+    });
+  }
+  /**
+ * groupClickHandler gets called from Group component when a groupname is clicked on
+ * It fires the groupSelected action
+ * @param {integer} groupId
+ * @return {*} void
+ */
+  groupClickHandler(groupId) {
+    /**
+ * calls the groupSelected action and passes the groupId of the group selected to it
+ * This action saves the id to the reducer
+ */
+    this.props.groupSelected(groupId);
+  }
+  /**
+   * render
+   * @return {*} li
+   */
+  render() {
+    $('.tooltipped').tooltip();
+    const listItems = this.props.groupList.map((group, index) => {
+      const selected = (group.id === Number(this.props.selectedGroup));
+      return (
+        <li className="each-group" key={index}>
```

Comment:
Eslint is throwing this warning because it sees the use of index as an anti-pattern and this might break your code application. 

A solution should be giving a unique property to your items which should be `id`.

Reply:
I tried using id but i was getting this error. But, i will look at it again and try to fix this. Thanks